### PR TITLE
Added support for AWS Trainium accelerator

### PR DESCRIPTION
**Pull request by mmcclean-aws**
Added support for the AWS Trainium Accelerator

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py [hunk 1]
```diff
@@ -287,6 +287,12 @@ def get_additional_columns(row) -> pd.Series:
             if row['InstanceType'] == 'p4de.24xlarge':
                 acc_name = 'A100-80GB'
                 acc_count = 8
+            if row['InstanceType'] == 'trn1.2xlarge':
```
Comment (cblmemo) [inline sky/clouds/service_catalog/data_fetchers/fetch_aws.py:290]: Shall we add a reference link for this instance type? Such as https://aws.amazon.com/ec2/instance-types/trn1/

**Review by cblmemo — COMMENTED** [review]
Thanks for contributing to SkyPilot! It is exciting ; ) Left some comments 🙂
I tried
```bash
$ sky show-gpus Trainium
Resources 'Trainium' not found. Try 'sky show-gpus --all' to show available accelerators.
```
and got the error above (`Inferentia` seems to have the same problem). Could you resolve this problem?
Also, seems like the workflow has failed. Do we need to update the [skypilot-catalog](https://github.com/skypilot-org/skypilot-catalog) repo too? cc @Michaelvll for a look

File: sky/clouds/service_catalog/data_fetchers/fetch_aws.py (hunk 1 above)
Comment (mmcclean-aws) [reply sky/clouds/service_catalog/data_fetchers/fetch_aws.py:290]: yes, good idea

**Comment by mmcclean-aws** [comment]
@cblmemo I see the catalog csv file [here](https://github.com/skypilot-org/skypilot-catalog/blob/master/catalogs/v5/aws/vms.csv) is not up to date with the Trainium accelerator type. How can this be regenerated ?

**Comment by Michaelvll** [comment]
> @cblmemo I see the catalog csv file [here](https://github.com/skypilot-org/skypilot-catalog/blob/master/catalogs/v5/aws/vms.csv) is not up to date with the Trainium accelerator type. How can this be regenerated ?

Hi @mmcclean-aws, sorry for the delay! The catalog file is generated by the command: `python -m sky.clouds.service_catalog.data_fetchers.fetch_aws`

If you would like to update the information for the trainium, it would be nice to update the file https://github.com/skypilot-org/skypilot/blob/master/sky/clouds/service_catalog/data_fetchers/fetch_aws.py accordingly.

**Review by Michaelvll — APPROVED** [review]
I just merged the change of `fetch_aws.py` change into the master first so that our catalog can be automatically updated with trainium. Merging this PR now. Thanks for adding the support for Trainium @mmcclean-aws!

**Pull request merged**